### ISSUE TITLE
Preparation for conversion to parameter groups

### ISSUE THREAD

**Issue by martinbudden** [iNavFlight/inav#472]
This issue is a task list of the things that need to be done before the conversion to parameter groups can begin. These are non-disruptive changes that can be implemented in their own right, but put the code into a position where the conversion to parameter groups can happen in one or two evenings of solid work.
- [x] 1. Split `config.c` and `config.h` into their constituent parts, as in Cleanflight
  - [x] a. Split eeprom code out of `config.c` PR https://github.com/iNavFlight/inav/pull/580
  - [x] b. Split features out of `config.c` PR https://github.com/iNavFlight/inav/pull/572
- [x] 2. Add in `streambuf` code
- [x] 3. Split MSP into its constituent parts, as in Cleanflight PR https://github.com/iNavFlight/inav/pull/571
- [x] 4. Convert MSP to use `streambuf` PR https://github.com/iNavFlight/inav/pull/645 PR https://github.com/iNavFlight/inav/pull/648
- [x] 5. Add in `parameter_group` code PR https://github.com/iNavFlight/inav/pull/581
- Convert to parameter groups - separate task, issue https://github.com/iNavFlight/inav/issues/473

See also issue https://github.com/betaflight/betaflight/issues/941 and PR https://github.com/cleanflight/cleanflight/pull/2034

**Comment by digitalentity** [comment]
Last one is going to be a one hell of a challenge, we'll introduce zillions of bugs on the way :laughing:

**Comment by martinbudden** [comment]
Yes, the last one will be the big one.

I've held off doing any work on this, since some of the changes in file structure/ directory structure although safe in themselves are a bit disruptive to the code. I thought it might obscure the work on the memory corruption bug.

Do you think it's OK to proceed now?

**Comment by digitalentity** [comment]
Let's branch master -> development and start working on PGs there, I want complete feature freeze in master until 1.2.

**Comment by martinbudden** [comment]
OK, sounds good.

No rush to create a development branch - I could do with a little break from this anyway. I'll be glad when it's all done, so I can spend more time implementing new stuff.

**Comment by digitalentity** [comment]
Looks like this is finished. Closing.
